test-src/org/pentaho/platform/dataaccess/datasource/wizard/csv/CsvUtilsTest.java

```diff
@@ -142,4 +161,168 @@ public void generateFields_EscapesColumnsNames() throws Exception {
+    assertEquals( 4, columnInfo.getPrecision() );
+    assertEquals( 7, columnInfo.getLength() );
+  }
+
+  @Test
+  public void testDateFormat1() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "20000229" ), "yyyyMMdd" );
+  }
+
+  @Test
+  public void testDateFormat2() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-20" ), "dd-MM-yy" );
+  }
+
+  @Test
+  public void testDateFormat3() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-2000" ), "dd-MM-yyyy" );
+  }
+
+  @Test
+  public void testDateFormat4() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "02/29/20" ), "MM/dd/yy" );
+  }
+
+  @Test
+  public void testDateFormat5() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "02/29/2000" ), "MM/dd/yyyy" );
+  }
+
+  @Test
+  public void testDateFormat6() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "2000-02-29" ), "yyyy-MM-dd" );
+  }
+
+  @Test
+  public void testDateFormat7() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "2000/02/29" ), "yyyy/MM/dd" );
+  }
+  //minutes
+  @Test
+  public void testDateFormat1m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "200002290000" ), "yyyyMMddHHmm" );
+  }
+
+  @Test
+  public void testDateFormat1m2() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "20000229 0000" ), "yyyyMMdd HHmm" );
+  }
+
+  @Test
+  public void testDateFormat2m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-20 00:00" ), "dd-MM-yy HH:mm" );
+  }
+
+  @Test
+  public void testDateFormat3m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-2000 00:00" ), "dd-MM-yyyy HH:mm" );
+  }
+
+  @Test
+  public void testDateFormat4m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "02/29/20 00:00" ), "MM/dd/yy HH:mm" );
+  }
+
+  @Test
+  public void testDateFormat5m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "02/29/2000 00:00" ), "MM/dd/yyyy HH:mm" );
+  }
+
+  @Test
+  public void testDateFormat6m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "2000-02-29 00:00" ), "yyyy-MM-dd HH:mm" );
+  }
+
+  @Test
+  public void testDateFormat7m() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "2000/02/29 00:00" ), "yyyy/MM/dd HH:mm" );
+  }
+
+  //seconds
+  @Test
+  public void testDateFormat1ms() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "20000229000000" ), "yyyyMMddHHmmss" );
+  }
+
+  @Test
+  public void testDateFormat1m2s() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "20000229 000000" ), "yyyyMMdd HHmmss" );
+  }
+
+  @Test
+  public void testDateFormat2ms() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-20 00:00:00" ), "dd-MM-yy HH:mm:ss" );
+  }
+
+  @Test
+  public void testDateFormat3ms() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "29-02-2000 00:00:00" ), "dd-MM-yyyy HH:mm:ss" );
+  }
+
+  @Test
+  public void testDateFormat4ms() {
+    testAssumeColumnDetails_Dates( Arrays.asList( "02/29/20 00:00:00" ), "MM/dd/yy HH:mm:ss" );
+  }
```

Comment:
What happens if the date is 02/10/12 00:00:00 ?
It will have same behavior such for  "02/29/20 00:00:00" from testDateFormat4ms.
I use this date just as an example of valid input.

Generally, How much of the behavior has changed ?
It just add a new format for date.

How many of these test cases have changed output with your change ?
It equals a count of new formats. 

Do they all run without your changes ?
No, they give wrong date format for date which format we do not have and the failures expected.

Were we outputing one format before your changes and another one after them ?
Yes. Now we detect format in most cases.